lib/bolt_ext/server.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+require 'sinatra'
+require 'bolt'
+require 'json'
+
+class TransportAPI < Sinatra::Base
+  set :server, :puma
+
+  post '/ssh/run_task' do
+    body = JSON.parse(request.body.read)
+    target = [Bolt::Target.new(body['target']['hostname'], body['target']['options'])]
+    task = Bolt::Task.new(body['task']['name'], body['task']['file'], body['task']['metadata'])
+    parameters = body['parameters']
+
+    @analytics = Bolt::Analytics.build_client
+    executor = Bolt::OrchExecutor.new(Bolt::Config.new, @analytics, false)
+
+    @r = ''
+    # Since this will only be on one node we can just set r to the result
+    executor.run_task(target, task, parameters) do |event|
+      if event[:type] == :node_result
+        @r = event[:result].to_json
+      end
+    end
+    response.body = @r
+    response.status = 200
+  end
+end
+
+TransportAPI.run!
```

Comment:
Maybe this should happen in `exe/bolt_server`? Seems weird to start a server when you require a file.